csharp/IceStorm/Weather/Station/Program.cs

```diff
@@ -56,3 +59,7 @@
 
 // Wait until the communicator is shut down. Here, this occurs when the user presses Ctrl+C.
 await communicator.shutdownCompleted;
+
+// Remove the subscription to the topic.
+await topic.unsubscribeAsync(weatherStation);
```

Comment:
We can mention the communicator shutdown doesn't affect the client-side functionality. I can look surprising to send a request after shutdown completed.